source/common/router/router.cc

```diff
@@ -386,11 +386,12 @@ Http::FilterHeadersStatus Filter::decodeHeaders(Http::HeaderMap& headers, bool e
   // Extract debug configuration from filter state. This is used further along to determine whether
   // we should append cluster and host headers to the response, and whether to forward the request
   // upstream.
-  const StreamInfo::FilterState& filter_state = callbacks_->streamInfo().filterState();
+  const StreamInfo::FilterStateSharedPtr& filter_state = callbacks_->streamInfo().filterState();
   const DebugConfig* debug_config =
-      filter_state.hasData<DebugConfig>(DebugConfig::key())
-          ? &(filter_state.getDataReadOnly<DebugConfig>(DebugConfig::key()))
-          : nullptr;
+      filter_state ? filter_state->hasData<DebugConfig>(DebugConfig::key())
```

Comment:
In many places in this change you have added a nullptr check where it did not exist before. This pointer should never be null, so you can remove this check here and elsewhere.

Reply:
I just feel better to have null protection for ptrs :) so that it doesn't bite later on...

removing it here..